scripts/spack/packages/serac/package.py

```diff
@@ -219,7 +219,7 @@ class Serac(CachedCMakePackage, CudaPackage):
     conflicts("cuda_arch=none", when="+cuda",
               msg="CUDA architecture is required")
     depends_on("amgx", when="+cuda")
-    cuda_deps = ["axom", "mfem", "raja", "sundials", "umpire"]
```

Comment:
raja, umpire, and tribol all have variants.

This should be broken apart into its own loop because the when should be `when=+tribol+cuda` for example

Reply:
i noticed this would be an issue as well but i guess i never created an issue on it.

what about sundials? if sundials were supposed to not build and `+cuda`, it would still depend on sundials

Reply:
ok i did pretty much what you said just also included sundials